cache_config.py

```diff
@@ -0,0 +1,216 @@
+    "settings_save":                {"type": "user",   "endpoints": ["settings_load"]},
+    "token_generate":               {"type": "user",   "endpoints": ["token_load"]},
+    "token_delete":                 {"type": "user",   "endpoints": ["token_load"]},
+}
+
+
+def authenticated() -> bool:
+    """Check if the user is authenticated.
+
+    :returns: True if the user is authenticated, False otherwise
+    """
+    return g.get("user") is not None and g.get("irods") is not None
+
+
+def get_user_identifier() -> str:
+    """Get user identifier generated from username, return unredacted username in development environments.
+
+    :returns: User identifier
+    """
+    if authenticated():
+        user = g.get('user')
+        is_development = app.config.get("YODA_ENVIRONMENT") == "development"
+        return user if is_development else hashlib.shake_256(user.encode("utf-8")).hexdigest(20)
+    else:
+        return "unauthenticated"
+
+
+def make_key(api_key: Optional[str] = None) -> str:
+    """Generate a cache key based on the request and authentication status.
+
+    :param api_key: Optional custom key identifying API endpoint. If None, defaults to request endpoint and method
+
+    :returns: A string representing the cache key
+    """
+    if api_key is None:
+        key = f"view-{session.sid}-{request.endpoint}_{request.method}"
+    else:
+        key = f"api-{api_key}"
+
+    user_identifier = get_user_identifier()
+
+    return f"{user_identifier}-{key}"
+
+
+def get_api_cache_functions() -> list:
+    """Return a list of function names from the API_CACHE_TIMEOUTS configuration."""
+    return list(API_CACHE_TIMEOUTS.keys())
+
+
+def get_api_cache_timeout(fn: str) -> int:
+    """Retrieve the cache timeout for a specific API function.
+
+    :param fn: The name of the API function
+
+    :returns: The cache timeout in seconds, or 0 if not found
+    """
+    return API_CACHE_TIMEOUTS.get(fn, {"timeout": 0})["timeout"]
+
+
+def filter_cache_keys(substring: str) -> List[str]:
+    """Retrieve cache keys that contain a specified substring.
+
+    :param substring: The substring to search for in cache key names
+
+    :returns: A list of matching cache keys
+    """
+    prefix = cache.cache.key_prefix
+    keys = cache.cache._write_client.keys(f"*{substring}*")
+    return [bs.decode("utf-8")[len(prefix):] for bs in keys]
+
+
+def clear_view_cache_keys(all_users: bool = False) -> None:
+    """Clear view cache keys associated with the current user or all users.
+
+    :param all_users: If True, clear view cache keys for all users
+    """
+    if all_users:
+        filter_key = "-view-"
+    else:
+        user_identifier = get_user_identifier()
+        filter_key = f"{user_identifier}-view-"
+
+    # Get the keys to delete.
+    keys_to_delete = [key for key in filter_cache_keys(filter_key)]
+    keys_to_delete = list(filter_cache_keys(filter_key))
```

Comment:
This suggestion did not appear as I expected, apologies. I meant that the line should be replaced. Let's try this again:
```suggestion
    keys_to_delete = list(filter_cache_keys(filter_key))
```